src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs

```diff
@@ -591,13 +591,13 @@ private CGSize PrepareLayoutInternal(bool createLayoutInfo, bool isCollectionCha
 				// We are layouting after an INotifyCollectionChanged operation(s). Cache the previous element sizes, under their new index 
 				// paths, so we can reuse them in order not to have to lay out elements with different databound sizes with their static size.
 				oldItemSizes = _itemLayoutInfos.SelectMany(kvp => kvp.Value)
-					.ToDictionary(
+					.ToDictionaryKeepLast(
 						kvp => OffsetIndexForPendingChanges(kvp.Key, NativeListViewBase.ListViewItemElementKind),
 						kvp => (CGSize?)kvp.Value.Size
 					);
 				oldGroupHeaderSizes = _supplementaryLayoutInfos
 					.UnoGetValueOrDefault(NativeListViewBase.ListViewSectionHeaderElementKind)?
-					.ToDictionary(
+					.ToDictionaryKeepLast(
```

Comment:
I see you also created a `ToDictionaryKeepFirst` but didn't use it, just making sure it was just to have it and that one of these calls wasn't supposed to be a `KeepFirst`. Makes sense to me as it is, just double checking.

Reply:
Just added for parity, in case we need the other impl. But, we dont know which is right, since there is no way to validate. And, it is also likely that neither are wrong or neither are correct.